### PR TITLE
#4181 Let linked-schema change work for BETA tablet

### PR DESCRIPTION
**Pull request by imay**

### CODE REVIEW

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -2141,7 +2141,7 @@ OLAPStatus SchemaChangeHandler::_parse_request(TabletSharedPtr base_tablet,
         *sc_directly = true;
     }
 
-    if (new_tablet->tablet_meta()->preferred_rowset_type() == BETA_ROWSET) {
+    if (base_tablet->tablet_meta()->preferred_rowset_type() != new_tablet->tablet_meta()->preferred_rowset_type()) {
```
Comment (kangkaisen) [inline be/src/olap/schema_change.cpp:2144]: The comment need update

**Comment by kangkaisen** [comment]
We should cover the schema_version_convert in UT, which is error-prone.

**Comment by morningman** [comment]
Please add an ISSUE to describe the problem you fixed.

**Comment by imay** [comment]
> Please add an ISSUE to describe the problem you fixed.

Issue is in the commit title 
#4181

**Review by kangkaisen — APPROVED** [review]
+1

**Review by chaoyli — APPROVED** [review]

**Comment by chaoyli** [comment]
LGTM

**Pull request merged**